test/test_utils.py

```diff
@@ -587,3 +590,51 @@ def test_abspath_no_symlink(self):
     def test_abspath_symlink(self, mock_islink):
         resolved_path = str(abspath("../gradio/gradio/test_data/lion.jpg"))
         assert ".." in resolved_path
+
+
+class TestGetTypeHints:
+    def test_get_type_hints(self):
+        class F:
+            def __call__(self, s: str):
+                return s
+
+        class C:
+            def f(self, s: str):
+                return s
+
+        def f(s: str):
+            return s
+
+        class GenericObject:
+            pass
+
+        test_objs = [F(), C().f, f]
+
+        for x in test_objs:
+            hints = get_type_hints(x)
+            assert len(hints) == 1
+            assert hints["s"] == str
+
+        assert len(get_type_hints(GenericObject())) == 0
+
+
+class TestCheckFunctionInputsMatch:
+    def test_check_function_inputs_match(self):
+        class F:
+            def __call__(self, s: str, evt: EventData):
+                return s
+
+        class C:
+            def f(self, s: str, evt: EventData):
+                return s
+
+        def f(s: str, evt: EventData):
+            return s
+
+        test_objs = [F(), C().f, f]
+
+        with warnings.catch_warnings():
```

Comment:
This test isn't testing anything I thin. Is the intention to check that no warnings are being emitted?

Reply:
Yes, this test will fail if the previous code is used.

Reply:
Should be good to go once the CHANGELOG is fixed